chia/wallet/cat_wallet/cat_wallet.py

```diff
@@ -600,39 +612,40 @@ async def create_tandem_xch_tx(
                 coins=chia_coins,
                 origin_id=origin_id,  # We specify this so that we know the coin that is making the announcement
                 negative_change_allowed=False,
-                coin_announcements_to_consume={announcement_to_assert} if announcement_to_assert is not None else None,
+                coin_announcements_to_consume=announcements_to_assert if announcements_to_assert is not None else None,
                 reuse_puzhash=reuse_puzhash,
             )
             assert chia_tx.spend_bundle is not None
-
-            message = None
-            for spend in chia_tx.spend_bundle.coin_spends:
-                if spend.coin.name() == origin_id:
-                    conditions = spend.puzzle_reveal.to_program().run(spend.solution.to_program()).as_python()
-                    for condition in conditions:
-                        if condition[0] == ConditionOpcode.CREATE_COIN_ANNOUNCEMENT:
-                            message = condition[1]
-
-            assert message is not None
-            announcement = Announcement(origin_id, message)
         else:
             chia_coins = await self.standard_wallet.select_coins(
                 fee,
                 min_coin_amount=min_coin_amount,
                 max_coin_amount=max_coin_amount,
                 excluded_coin_amounts=excluded_coin_amounts,
             )
+            origin_id = list(chia_coins)[0].name()
             selected_amount = sum([c.amount for c in chia_coins])
             chia_tx = await self.standard_wallet.generate_signed_transaction(
                 uint64(selected_amount + amount_to_claim - fee),
                 (await self.standard_wallet.get_puzzle_hash(not reuse_puzhash)),
                 coins=chia_coins,
                 negative_change_allowed=True,
-                coin_announcements_to_consume={announcement_to_assert} if announcement_to_assert is not None else None,
+                coin_announcements_to_consume=announcements_to_assert if announcements_to_assert is not None else None,
```

Comment:
what's the point of the `if not None else None`?

Reply:
The default value in `generate_signed_transaction` for this parameter is `None`.  The `is not None else None` is basically just choosing not to set the announcements if we didn't receive any input to set them.